homework/03LessonThree.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+
+var assert = require('assert');
+
+// ****
+// Scope
+// ****
+
+// DO NOT MODIFY
+var lastName = 'plaid';
+
+// DO NOT MODIFY
+function printFullName() {
+  var firstName = 'bob';
+  return firstName + lastName;
+}
+
+// DO NOT MODIFY
+var firstName = 'susan';
+
+// Problem 1:
+// What is the value of firstName at this point in the file?
+var valueOfFirstName = 'susan';
+
+// Problem 2:
+// What is the value of lastName at this point in the file?
+var valueOfLastName = 'plaid';
+
+// Problem 3:
+// Fix the following function so that it returns the first argument
+function returnTheFirstArgument(firstArgument) {
+  return firstArgument;
+}
+
+// ****
+// Concept Checkpoint
+//
+// Write your answer in comments
+//
+// What is scope?
+//
+// Your Answer Goes Here:
+// The 'scope' of an item refers to where it can be accessed from.
+//
+
+
+// ****
+// Ternary Operator
+// ****
+
+// DO NOT MODIFY
+var ternaryResult1 = true   ? 'first' : 'second';
+var ternaryResult2 = false  ? 'first' : 'second';
+var ternaryResult3 = 4      ? 'first' : 'second';
+var ternaryResult4 = ""     ? 'first' : 'second';
+
+// Problem 4:
+// What is the value of ternaryResult1
+var problemFourAnswer = 'first';
+
+// Problem 5:
+// What is the value of ternaryResult2
+var problemFiveAnswer = 'second';
+
+// Problem 6:
+// What is the value of ternaryResult3
+var problemSixAnswer = 'first';
+
+// Problem 7:
+// What is the value of ternaryResult4
+var problemSevenAnswer = 'second';
+
+// ****
+// Concept Checkpoint
+//
+// Write your answer in comments
+//
+// What is the ternary operator?  How does it differ from normal conditional statements?
+//
+// Your Answer Goes Here:
+// The ternary operator is used as a shortcut for the 'if' statement. It is shorter than normal conditional statements but serves the same functionality.
+//
+
+
+// ****
+// Arrays
+// ****
+
+// Problem 8:
+// Create an empty array
+var emptyArray = [ ];
+
+// Problem 9:
+// Create an array with 5 elements in it
+var lengthFiveArray = [1,2,3,4,5];
+
+// DO NOT MODIFY
+var nameArray = ['bob', 'fred', 'susan'];
+
+// Problem 10:
+// Replace the value 'fred' in nameArray with 'george'
+nameArray[1] = "george";
+// DO NOT MODIFY
+var threeByThreeArray = [
+  [1, 2, 3],
+  [4, 5, 6],
+  [7, 8, 9]
+];
+
+// Problem 11:
+// Replace the center element of threeByThreeArray, which is 5, with something else.
+threeByThreeArray[1][1] = 'Hello';
+// Problem 12:
+// create a 2 by 2 (2 rows, 2 columns) nested array
+var twoByTwoArray = [
+  [1,2],
+  [3,4]
+];
+
+// ****
```

Comment:
Great job knocking out this homework before the rest of the class! All your tests passed.

![crying](https://i.imgflip.com/18me26.jpg)